src/components/ProductsList/components/ProductCard/components/ProductCardDetails/ProductCardDetails.tsx

```diff
@@ -0,0 +1,41 @@
+import { MediaQueries, Routes, tempProductId } from "@/common/constants";
+import { NavigationLink, Product } from "@/common/types";
+import { Rating } from "@/components/UI";
+import { useMatchMedia } from "@/hooks";
+import { useAppSelector } from "@/store";
+import { generateCharacteristics } from "@/utils";
+import { FC, useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { ProductCardDetailsDesktop, ProductCardDetailsMobile } from "./components";
+import styles from "./ProductCardDetails.module.scss";
+
+export const ProductCardDetails: FC<Product> = ({ title, description, rating, originCountry, brand, delivery, stock }) => {
+  const isMobile = useMatchMedia(`(max-width: ${MediaQueries.TABLET}px)`);
+  const { isPending } = useAppSelector(state => state.products);
+  const [productCharacteristics, setProductCharacteristics] = useState<NavigationLink[]>([]);
+
+  useEffect(() => {
+    if (!isPending) {
+      setProductCharacteristics(generateCharacteristics({ originCountry, brand, delivery, stock }));
+    }
+  }, []);
```

Comment:
Why do you need to do that ?
In React, it is generally not recommended to set the props directly to the state of a component, it can lead to unexpected behavior and bugs in your application.